packages/paste-website/src/component-examples/ChatLogExamples.ts

```diff
@@ -311,3 +311,66 @@ render(
   <SampleChat />
 )
 `.trim();
+
+export const chatLoggerExample = `
+const ChatLoggerExample = () => {
+  const [templateIdx, setTemplateIdx] = React.useState(2);
+  const { chats, push, pop } = useChatLogger(
+    chatFactory(chatTemplates[0]),
+    chatFactory(chatTemplates[1])
+  );
+
+  const pushChat = () => {
+    const template = chatTemplates[templateIdx];
+    push(chatFactory(template));
+    setTemplateIdx((idx) => ++idx % chatTemplates.length);
+  }
+
+  const popChat = () => {
+    pop();
+    setTemplateIdx((idx) => idx === 0 ? idx : --idx % chatTemplates.length);
+  }
+
+  return(
+    <Stack orientation="vertical">
+      <ButtonGroup>
+        <Button variant="primary" onClick={pushChat}>
+          Push Chat
+        </Button>
+        <Button variant="primary" onClick={popChat}>
+          Pop Chat
+        </Button>
+      </ButtonGroup>
+      <ChatLogger chats={chats} />
+    </Stack>
+  )
+}
+
+const chatFactory = ([ message, variant, metaLabel, meta ]) => {
+  const time = new Date().toLocaleString(
+    'en-US',
+    { hour: 'numeric', minute: "numeric", hour12: true }
+  )
+
+  return {
+    variant,
+    content: (
+      <ChatMessage variant={variant}>
+        <ChatBubble>{message}</ChatBubble>
+        <ChatMessageMeta aria-label={metaLabel + time}>
+          <ChatMessageMetaItem>{meta + time}</ChatMessageMetaItem>
+        </ChatMessageMeta>
+      </ChatMessage>
+    )
+  }
+};
+
+const chatTemplates = [
```

Comment:
nit (not-blocking): could we move this and chatFactory to above the component "ChatLoggerExample"? More readable this way